src/components/PostDetails.tsx

```diff
@@ -1,106 +1,122 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { deleteComment, getPostComments, postComment } from '../api/comments';
+import { Comment, CommentData } from '../types/Comment';
+
+type Props = {
+  post: Post;
+};
+
+export const PostDetails: React.FC<Props> = ({ post }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [hasError, setHasError] = useState('');
```

Comment:
Consider initializing `hasError` as a boolean or an object to allow for more flexible error handling in the future.